src/types.ts

```diff
@@ -4,7 +4,7 @@ export type ID = string
 export type AccessToken = string
 
 export interface ConstructorOptions {
-  token: () => AccessToken
+  token: () => Promise<AccessToken>
```

Comment:
```suggestion
  token: () => Promise<AccessToken> | AccessToken
```

Reply:
should I do this for backward compatibility?

Reply:
await simply returns non Promise value, but Typescript complains if token() returns non Promise value.

Reply:
Yes, I'll add it for the release

Reply:
I just released 1.2.1 that should fix this